### ISSUE TITLE
Is there anyway to speed up learning with gnu?

### ISSUE THREAD

**Issue by jyoungrok** [tflearn/tflearn#440]
Is there anyway to speed up learning?
My program seems to stop  for a while  at the moment when one epoch finished
I guess it is because that it is little slow to calculate validation accuracy
However, I do not know why it is so slow compared to time learning for only one epoch
Is there any way to speed up with calculating validation accuracy each epoch?

**Comment by burness** [comment]
Could you share your code and the log? I don't think the validation accuracy will take too much time because it  is only a forward computation

**Comment by jyoungrok** [comment]
```
   def learning(self,id,numHidden=100,batchSize=100,epoch=100,storePath=None):

     net = input_data(shape=[None, self.trainingData.maxDataLength, self.trainingData.numFeatures])
        # net = embedding(net, input_dim=20000, output_dim=128)
        net = tflearn.lstm(net, numHidden, dropout=0.8)
        net = dropout(net, 0.5)
        net = fully_connected(net, 2, activation='softmax')
        net = regression(net, optimizer='adam', loss='categorical_crossentropy')


        # Training
        model = tflearn.DNN(net, tensorboard_verbose=0,tensorboard_dir=LOG_DIRECTORY,best_checkpoint_path=storePath)


        model.fit(self.trainingData.data, self.trainingData.label,
                      snapshot_step=10 * math.ceil(len(self.trainingData.data) / batchSize), n_epoch=epoch,
                      validation_set=0.1, show_metric=True,
                      batch_size=batchSize, run_id=id)

```

The above is my code
It is not much different from tutorial code

```
Training Step: 22330  | total loss: 0.47091
| Adam | epoch: 154 | loss: 0.47091 - acc: 0.7579 | val_loss: 0.44750 - val_acc: 0.7628 -- iter: 144645/144645
--
Training Step: 22475  | total loss: 0.46667
| Adam | epoch: 155 | loss: 0.46667 - acc: 0.7596 | val_loss: 0.44679 - val_acc: 0.7634 -- iter: 144645/144645
--
Training Step: 22540  | total loss: 0.46403
| Adam | epoch: 156 | loss: 0.46403 - acc: 0.7609 | val_loss: 0.44721 - val_acc: 0.7616 -- iter: 065000/144645
--
```
And these are log samples

Whenever a line of these log had just been printed, another line of log started being printed after few seconds (about 1s)
It seems to do something at this moment
